xpublish_edr/plugin.py

```diff
@@ -2,19 +2,34 @@
 OGC EDR router for datasets with CF convention metadata
 """
 import logging
-from typing import List, Optional
+from functools import cache
+from typing import Hashable, List, Optional, Tuple
 
+import cachey
+import dask
 import pkg_resources
 import xarray as xr
-from fastapi import APIRouter, Depends, HTTPException, Request
+from fastapi import APIRouter, Depends, HTTPException, Request, Response
 from xpublish import Dependencies, Plugin, hookimpl
+from xpublish.utils.cache import CostTimer
 
 from .formats.to_covjson import to_cf_covjson
 from .query import EDRQuery, edr_query, edr_query_params
 
 logger = logging.getLogger("cf_edr")
 
 
+def cache_key_from_request(
+    route: str,
+    request: Request,
+    query: EDRQuery,
+    dataset: xr.Dataset,
+) -> Tuple[Hashable, ...]:
+    """Generate a cache key from the request and query parameters"""
+    return (route, request, query, dask.base.tokenize(dataset))
```

Comment:
per https://github.com/pydata/xarray/issues/4738#issuecomment-1139544716, we may need to set some dask config to get this to be deterministic: `with dask.config.set({"tokenize.ensure-deterministic":True}):`